components/formats-bsd/src/loci/formats/in/FakeReader.java

```diff
@@ -986,7 +990,18 @@ private void fillRegions(MetadataStore store, int imageIndex) {
         roiID = ROI_PREFIX + roiCount;
         store.setROIID(roiID, roiCount);
         store.setMaskID(SHAPE_PREFIX + roiCount, roiCount, 0);
+        store.setMaskX((double)ROI_SPACING, roiCount, 0);
+        store.setMaskY((double)ROI_SPACING, roiCount, 0);
+        store.setMaskWidth((double)ROI_SPACING, roiCount, 0);
+        store.setMaskHeight((double)ROI_SPACING, roiCount, 0);
         store.setImageROIRef(roiID, imageIndex, roiRefCount);
+        byte[] rawBytes = new byte[ROI_SPACING*ROI_SPACING];
+        random.nextBytes(rawBytes);
+        String base64encoded = DatatypeConverter.printBase64Binary(rawBytes);
+        byte[] binData = base64encoded.getBytes(StandardCharsets.UTF_8);
+        store.setMaskBinData(binData, roiCount, 0);
```

Comment:
Why are we dealing with Base64 encoding at this level?  Why isn't the API dealing solely with raw bytes?

Should this conversion to/from Base64 not be done internally by the BinData object, either in the getter/setter or when serialising to/from XML?